src/js/components/SchemaForm.js

```diff
@@ -99,12 +99,14 @@ class SchemaForm extends mixin(StoreMixin, InternalStorageMixin) {
     // Default method.
   }
 
-  handleAddRow(prop, definition, newDefinition) {
+  handleAddRow(prop, definition, newDefinition, index) {
     let propID = Util.uniqueID(prop);
     newDefinition = FormUtil.getMultipleFieldDefinition(
       prop,
       propID,
-      newDefinition
+      newDefinition,
+      undefined,
+      index
```

Comment:
In a separate PR, I would love for us to pass an object with these properties, rather than carefully construct this list of arguments each time we call the `#getMultipleFieldDefinition` method. But again, that's not in scope of this PR!